src/vote_stage.rs

```diff
@@ -390,6 +389,11 @@ pub mod tests {
 
         // leader should vote now
         assert!(vote_blob.is_ok());
+
+        // vote should be valid
+        let blob = vote_blob.unwrap().pop_front().unwrap();
+        let tx = deserialize(&(blob.read().unwrap().data)).unwrap();
+        assert!(bank.process_transaction(&tx).is_ok());
```

Comment:
you're saying this assert!() failed unless you took out the front-run at line 145?  I don't understand this code, yet.

Reply:
yep, it registered the signature, then sent the vote to itself...  @sakridge and I had a miscommunication. I was expecting the code to inject the vote into the record stage and skip the network hop

Reply:
OK. i see so the leader's TPU is the blob 'to' address, but instead of doing that we would inject directly into the record stage.